frontend/apps/pablo/defi/hooks/swaps/useSwaps.ts

```diff
@@ -129,13 +129,13 @@ export function useSwaps({ selectedAccount }: { selectedAccount?: InjectedAccoun
     }
   }, [selectedPool, updateSpotPrice]);
 
-  const { constantProductPools } = usePoolsSlice();
+  const { liquidityPools } = usePoolsSlice();
   useEffect(() => {
-    if (constantProductPools.length > 0) {
-      const pool = constantProductPools.find((_constantPool) => {
-        const pair = _constantPool.getPair();
-        const pairBase = pair.getBaseAsset().toString();
-        const pairQuote = pair.getQuoteAsset().toString();
+    if (liquidityPools.length > 0) {
+      const pool = liquidityPools.find((_constantPool) => {
+        const pair = Object.values(_constantPool.getAssets().assets);
+        const pairBase = pair[0].toString();
```

Comment:
Will pair always be defined?

Reply:
yup @andresmechali as long as pool data is correct from chain